Makefile

```diff
@@ -56,6 +56,11 @@ test: off
 test-integration: build-ci
 	cd tests/integration && go test -race ./...
 
+# This is needed for osx because this os does not support static linking
+# Use the build target without the static flag
+test-integration-osx: build
+	cd tests/integration && go test -race ./...
```

Comment:
i would prefer to keep `test-integration` intact and detect the os instead 

```bash
if [[ $OSTYPE == 'darwin'* ]]; then
  ...
fi
```

Reply:
as this is used in CI, i would like to keep it separate.